internal/provider/resource_environment.go

```diff
@@ -365,6 +342,50 @@ func createEnvironmentRequestFromData(ctx context.Context, data *environmentReso
 	return environmentRequest, nil
 }
 
+func (r *environmentResource) processProperties(ctx context.Context, req resource.UpdateRequest, data environmentResourceData) map[string]interface{} {
+	var oldPropertiesState map[string]string
+	req.State.GetAttribute(ctx, path.Root("properties"), &oldPropertiesState)
+
+	properties := make(map[string]interface{})
+	// Send `properties = nil` to API if in configuration `properties = nil`, `properties = {}` or NO properties
+	if data.Properties.IsNull() || data.Properties.IsUnknown() || len(data.Properties.Elements()) == 0 {
+		return nil
+	}
```

Comment:
Is this the main change that leads to TF detection of the absence of properties?

Reply:
This line sends `properties=nil` to API to delete properties for environment. It does it if in configuration `properties = nil`, `properties = {}` or NO properties